compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1621,7 +1621,7 @@ public BType readType(int cpI) throws IOException {
                     int funcCount = inputStream.readInt();
                     for (int i = 0; i < funcCount; i++) {
                         //populate intersection type object functions
-                        if (isImmutable(objectSymbol.flags) && Symbols.isFlagOn(flags, Flags.ANONYMOUS)) {
+                        if (isImmutable(objectSymbol.flags)) {
```

Comment:
We can move the logic out of for loop later.

Reply:
Updated [b6a0152](https://github.com/ballerina-platform/ballerina-lang/pull/37339/commits/b6a01528f6cf6a634662421b00165834fdfb76b9)